aspnetcore/tutorials/first-mvc-app/adding-model.md

```diff
@@ -19,6 +19,298 @@ The model classes you create are known as POCO classes (from **P**lain **O**ld *
+
+The automatic creation of the database context and [CRUD](https://wikipedia.org/wiki/Create,_read,_update_and_delete) (create, read, update, and delete) action methods and views is known as *scaffolding*.
+
+### [Visual Studio Code](#tab/visual-studio-code) 
+
+* Open a command window in the project directory (The directory that contains the *Program.cs*, *Startup.cs*, and *.csproj* files).
+* On Linux, export the scaffold tool path:
+
+  ```console
+    export PATH=$HOME/.dotnet/tools:$PATH
+  ```
+
+* Run the following command:
+
+  ```console
+     dotnet aspnet-codegenerator controller -name MoviesController -m Movie -dc MvcMovieContext --relativeFolderPath Controllers --useDefaultLayout --referenceScriptLibraries
+  ```
+
+[!INCLUDE [explains scaffold generated params](~/includes/mvc-intro/model4.md)]
+
+### [Visual Studio for Mac](#tab/visual-studio-mac)
+
+* Open a command window in the project directory (The directory that contains the *Program.cs*, *Startup.cs*, and *.csproj* files).
+
+* Run the following command:
+
+  ```console
+     dotnet aspnet-codegenerator controller -name MoviesController -m Movie -dc MvcMovieContext --relativeFolderPath Controllers --useDefaultLayout --referenceScriptLibraries
+  ```
+
+[!INCLUDE [explains scaffold generated params](~/includes/mvc-intro/model4.md)]
+---
+
+<!-- End of tabs                  -->
+
+If you run the app and click on the **Mvc Movie** link, you get an error similar to the following:
+
+``` error
+An unhandled exception occurred while processing the request.
+
+SqlException: Cannot open database "MvcMovieContext-<GUID removed>" requested by the login. The login failed.
+Login failed for user 'Rick'.
+
+System.Data.SqlClient.SqlInternalConnectionTds..ctor(DbConnectionPoolIdentity identity, SqlConnectionString
+```
+
+You need to create the database, and you use the EF Core [Migrations](xref:data/ef-mvc/migrations) feature to do that. Migrations lets you create a database that matches your data model and update the database schema when your data model changes.
+
+<a name="pmc"></a>
+
+## Initial migration
+
+In this section, the following tasks are completed:
+
+* Add an initial migration.
+* Update the database with the initial migration.
+
+# [Visual Studio](#tab/visual-studio)
+
+1. From the **Tools** menu, select **NuGet Package Manager** > **Package Manager Console** (PMC).
+
+   ![PMC menu](~/tutorials/first-mvc-app/adding-model/_static/pmc.png)
+
+1. In the PMC, enter the following commands:
+
+   ```console
+   Add-Migration InitialCreate
+   Update-Database
+   ```
+
+    `Add-Migration InitialCreate`: Generates an *Migrations/{timestamp}_InitialCreate.cs* migration file. Because this is the first migration, the generated class contains code to create the database schema.
+    `Update-Database`: Updates the database to the last migration. The last migration is the `InitialCreate`  class created with the previous command.
```

Comment:
Consider going with "latest" here and in the following sentence instead of "last".